prov/verbs/src/fi_verbs.c

```diff
@@ -943,24 +968,30 @@ static int fi_ibv_getinfo(uint32_t version, const char *node, const char *servic
 {
 	struct rdma_cm_id *id;
 	struct rdma_addrinfo *rai;
-	struct fi_info *fi;
+	struct fi_info *fi, *_info;
 	int ret;
 
 	ret = fi_ibv_init_info(hints);
 	if (ret)
 		return ret;
 
-	/* TODO When multiple devices are present verbs_info would be list
-	 * of info. In that case we have to choose the correct verbs_info to 
-	 * check against if we have a valid ibv_context in id->verbs */
+	assert(verbs_info);
+
 	ret = fi_ibv_create_ep(node, service, flags, hints, &rai, &id);
 	if (ret)
 		return ret;
 
-	assert(!id->verbs || !strcmp(verbs_info->domain_attr->name,
-				     ibv_get_device_name(id->verbs->device)));
+	if (id->verbs) {
+		for (_info = verbs_info; _info; _info = _info->next) {
+			if (!strcmp(_info->domain_attr->name,
+				ibv_get_device_name(id->verbs->device)))
+				break;
+		}
+	} else {
+		_info = verbs_info;
+	}
```

Comment:
I wonder if there should be some generic info structure that isn't associated with a specific device.  Otherwise, the default settings are whatever the first device has.  This may be a separate change.